src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -51,37 +51,60 @@ import {
 } from '@patternfly/react-core';
 
 import AuthorizedSSHKey from './AuthorizedSSHKey';
+import { useDrawerContext } from './useDrawerContext';
+import {
+  doNameParameterExists,
+  getTemplateNameParameterValue,
+  setTemplateParameterValue,
+  uploadFiles,
+} from './utils';
 
 type TemplatesCatalogDrawerCreateFormProps = {
   authorizedSSHKey: string;
   canQuickCreate: boolean;
-  initialVMName?: string;
-  isBootSourceAvailable: boolean;
   namespace: string;
   onCancel: () => void;
   subscriptionData: RHELAutomaticSubscriptionData;
-  template: V1Template;
 };
 
 export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFormProps> = memo(
-  ({
-    authorizedSSHKey,
-    canQuickCreate,
-    initialVMName,
-    isBootSourceAvailable,
-    namespace,
-    onCancel,
-    subscriptionData,
-    template,
-  }) => {
+  ({ authorizedSSHKey, canQuickCreate, namespace, onCancel, subscriptionData }) => {
     const history = useHistory();
     const { t } = useKubevirtTranslation();
-    const { updateTabsData, updateVM, vm } = useWizardVMContext();
 
-    const [vmName, setVMName] = useState(initialVMName || '');
+    const { updateTabsData, updateVM } = useWizardVMContext();
+
+    const {
+      cdFile,
+      diskFile,
+      isBootSourceAvailable,
+      setTemplate,
+      template,
+      uploadCDData,
+      uploadDiskData,
+      vm,
+    } = useDrawerContext();
+
+    const nameParameterExists = doNameParameterExists(template);
+
+    const [vmName, setVMName] = useState(generateVMName(template));
+
+    const nameField = nameParameterExists
+      ? getTemplateNameParameterValue(template) || vmName
+      : vmName;
```

Comment:
?

Reply:
removed the `|| vmName` part.

This says: if the template has the name parameter, show this and if the user change the name using the input, change the template parameter. 
If the template does not have the name parameter, generate a name, and if the user changes it, change just that state